modules/elasticsearch-index/src/main/java/org/opencastproject/elasticsearch/index/ElasticsearchIndex.java

```diff
@@ -130,15 +139,39 @@ public void activate(ComponentContext ctx) throws ComponentException {
   /**
    * OSGi callback to deactivate this component.
    *
-   * @param ctx
-   *          the component context
    * @throws IOException
+   *          If closing the index fails
    */
   @Deactivate
-  public void deactivate(ComponentContext ctx) throws IOException {
+  public void deactivate() throws IOException {
     close();
   }
 
+  /**
+   * OSGi callback for configuration changes.
+   *
+   * @param properties
+   *          The configuration
+   */
+  @Modified
+  public void modified(Map<String, Object> properties) {
+    super.modified(properties);
+
+    maxRetryAttemptsGet = NumberUtils.toInt((String) properties.get(MAX_RETRY_ATTEMPTS_GET_PROPERTY),
+            DEFAULT_MAX_RETRY_ATTEMPTS_GET);
+    retryWaitingPeriodGet = NumberUtils.toInt((String) properties.get(RETRY_WAITING_PERIOD_GET_PROPERTY),
```

Comment:
I have to be that guy, but `max.waiting.*=-1` does bad things since Thread.sleep throws an `IllegalArgumentException` for a negative waiting time.

Further objections: While *technically* correct in that a maximum retry count of -1 never retries, I'm pretty sure we should throw an exception, or at least warn that this is a dumb idea.  Also, setting non-integer values yeilds 0s, but I'm not sure there's much you can do with that since you're using NumberUtils.